c10/core/Storage.h

```diff
@@ -12,12 +12,12 @@ struct C10_API Storage {
   // Allocates memory buffer using given allocator and creates a storage with it
   Storage(
       caffe2::TypeMeta data_type,
-      size_t size,
+      size_t size_bytes,
```

Comment:
Changing this constructor in a semantic way without changing the type signature looks really error prone (as you found out empirically!) My advice to you would have been to either (1) NOT change the meaning of this constructor; i.e., we use the passed in data type to compute what the bytes are, or (2) remove this constructor entirely, and replace with some new constructors that specifically are tagged to be "Bytes" or "Items". You can use the class enum factory trick; check out DispatchKeySet.h for an example

Reply:
I particularly don't like option 1 because I'm going to remove the data type for the next PR. I guess I could have just waited until then to change the constructor's element count to a byte count though.

I think I understand the idea behind the class enum factory, but let me just make sure. Basically, you just create a new enum type (with one possible value) and require the one of the constructor arguments (probably the first argument makes the most sense) to be that enum type. Is that all it is? Seems like that would be better than option 1 at this point.

I wonder if the compiler would optimize out the enum argument since its always going to be the same value.

Reply:
Yep, that's right. If the constructor gets inlined it should get optimized away; easy enough to check in godbolt.

Reply:
godbolt shows that without any optimization flags, there's just one extra `mov` instruction from the enum argument. But with any amount of optimization, `-O1` or greater, a constructor with the enum argument has the same instructions as a constructor without the enum. Looks like pytorch uses `-O2`, at least in my environment, so it will be optimized out.

Even though I'm close to having all the CI tests passing, I'll go ahead and make this change because it will probably help me find the remaining bugs. I guess it's also slightly possible that some Storage constructor somewhere isn't getting exercised by the unit tests, so it would be good to find out for sure.